src/main/java/calculator/validator/InputValidator.java

```diff
@@ -0,0 +1,20 @@
+package calculator.validator;
+
+public class InputValidator {
+    private final ValueValidator valueValidator = new ValueValidator();
+    private final DelimiterValidator delimiterValidator = new DelimiterValidator();
+    private final NumberValidator numberValidator = new NumberValidator();
+
+    // 전체 입력 유효성 검증
+    public boolean isValidInput(String input) {
+        if (!valueValidator.isLastDigit(input)) {
+            return false; // 마지막 문자가 숫자가 아닌 경우
+        }
+
+        if (delimiterValidator.containsInvalidDelimiters(input)) {
+            return false; // 잘못된 구분자가 있는 경우
+        }
+
+        return !numberValidator.containsInvalidCharacters(input); // 숫자 검증
```

Comment:
```suggestion

        return numberValidator.containsOnlyValidCharacters(input);
```

부정문이 많으면 로직을 읽기가 어려워질 것 같은데, 어떻게 생각하시나요?

Reply:
구현하면서도 뭔가 찜찜했는데 확실히 bool타입의 경우 보기쉽게 가공하는 편이 가독성이 좋을 것 같습니다!